src/qt/psbtoperationsdialog.cpp

```diff
@@ -0,0 +1,265 @@
+    tx_description.append(" * ");
+    if (!*analysis.fee) {
+        // This happens if the transaction is missing input UTXO information.
+        tx_description.append(tr("Unable to calculate transaction fee or total transaction amount."));
+    } else {
+        tx_description.append(tr("Pays transaction fee: "));
+        tx_description.append(BitcoinUnits::formatWithUnit(BitcoinUnits::BTC, *analysis.fee));
+
+        // add total amount in all subdivision units
+        tx_description.append("<hr />");
+        QStringList alternativeUnits;
+        for (const BitcoinUnits::Unit u : BitcoinUnits::availableUnits())
+        {
+            if(u != m_client_model->getOptionsModel()->getDisplayUnit())
+                alternativeUnits.append(BitcoinUnits::formatHtmlWithUnit(u, totalAmount));
+        }
+        tx_description.append(QString("<b>%1</b>: <b>%2</b>").arg(tr("Total Amount"))
+            .arg(BitcoinUnits::formatHtmlWithUnit(m_client_model->getOptionsModel()->getDisplayUnit(), totalAmount)));
+        tx_description.append(QString("<br /><span style='font-size:10pt; font-weight:normal;'>(=%1)</span>")
+            .arg(alternativeUnits.join(" " + tr("or") + " ")));
+    }
+
+    size_t num_unsigned = CountPSBTUnsignedInputs(psbtx);
+    if (num_unsigned > 0) {
+        tx_description.append("<br><br>");
+        tx_description.append(tr("Transaction has %1 unsigned inputs.").arg(QString::number(num_unsigned)));
+    }
+
+    return tx_description.toStdString();
+}
+
+void PSBTOperationsDialog::showStatus(const QString &msg, StatusLevel level) {
+    m_ui->statusBar->setText(msg);
+    switch (level) {
+        case StatusLevel::INFO: {
+            m_ui->statusBar->setStyleSheet("QLabel { background-color : lightgreen }");
+            break;
+        }
+        case StatusLevel::WARN: {
+            m_ui->statusBar->setStyleSheet("QLabel { background-color : orange }");
+            break;
+        }
+        case StatusLevel::ERR: {
+            m_ui->statusBar->setStyleSheet("QLabel { background-color : red }");
+            break;
+        }
+    }
+    m_ui->statusBar->show();
+}
+
+size_t PSBTOperationsDialog::couldSignInputs(const PartiallySignedTransaction &psbtx) {
+    size_t n_signed;
+    bool complete;
+    TransactionError err = m_wallet_model->wallet().fillPSBT(m_transaction_data, complete, SIGHASH_ALL, false /* sign */, false /* bip32derivs */, &n_signed);
+
+    if (err != TransactionError::OK) {
+        return 0;
+    }
+    return n_signed;
+}
+
+void PSBTOperationsDialog::showTransactionStatus(const PartiallySignedTransaction &psbtx) {
+    PSBTAnalysis analysis = AnalyzePSBT(psbtx);
+    size_t n_could_sign = couldSignInputs(psbtx);
+
+    switch (analysis.next) {
+        case PSBTRole::UPDATER: {
+            showStatus(tr("Transaction is missing some information about inputs."), StatusLevel::WARN);
+            break;
+        }
+        case PSBTRole::SIGNER: {
```

Comment:
When all inputs have `"next": "finalizer"` the transaction itself may still be `"next": "signer"`. This seems to be a bug (@achow101 or feature?) in `analyzepsbt`. If it's a bug, we can ignore it for this PR, if it's a feature, we should handle it, though imo it can wait for a followup.

Reply:
I am going to assume this is a bug, so I'm happy to push to a followup. I have been tempted to try to rewrite the flow of AnalyzePSBT, with an eye to simplifying it and making its correctness easier to check.

Reply:
Sounds like a bug

Reply:
https://github.com/bitcoin/bitcoin/pull/18220